### DISCUSSION TITLE
Could bundlejs be a full blown editor.

### DISCUSSION THREAD

**Discussion by frog-o** [okikio/bundle#45] [Ideas]
I wanted to make an web editor using pyscript because I like python better 4 it readability. IMHO JavaScript is a bit of a pain.  I wanted the ide to feature 

- simple to install or no install (Just extract to dir or visit website)
- run anywhere easily (just click index.html)
- portable run on Linux, Mac or any os with a modern browser
- function with out Internet(eventually not needing it)
- customizable (support plugin with wasm)
- language independent(use as many wasm module as possible)
- git  bazaar or other svc support
- easily to read (source code to be simple)
- not a bunch of hidden config files 
- easily previews python,PHP and others(threw wasm)
- not depending  on python, PHP install on server
- everything running in web browser
- all config in config directory

It looks like you doing much of what i wanted to do already and you said you wanted to know if we like your project and our thoughts.
Here mine 

- I think what you doing is great
- i think it needs to be more than a bundler to gain attraction 🧲

If you don't mind i would like yours

- I like to know you're thought on my ide is feasible 

- There are many 3 main JavaScript text editors

1.) Ace
2.) codemirror
3.) The one you are using

Is there a reason for your choice?

IMHO I do too much research and don't get anything done. So i probably wouldn't ever be able to do the above IDE and i understand if you don't want to commit on this.

**Comment by okikio** [comment]
Sorry for the late response. Monaco is the same editor as the one vscode uses and it's feature rich that's the main reason I went for it, but do note Monaco is very inefficient e.g. if your lighthouse scores are 99, adding monaco will reduce it to ~47 if you're lucky. 

> If you can use Codemirror, I'd recommend doing that

Your goal is a python repl env, it's totally possible but you'd first need the python runtime WASM and potential a monaco python lsp (not sure if there are any out there), but yeah that's what you'd need 

This article may help https://www.theregister.com/2021/11/30/python_web_wasm/

**Comment by frog-o** [comment]
Pyscript i thought does that for you it beta but it kind of cool.  I done a
lot of research on the weekend and in the past i found another build system
that uses esbuild that was really cool if you're interested i can look for
the name, also tsx is really cool. I was trying to make a no build option
in someone else typescript project by use bable-standalone the problem i am having on that project was the
code scattered all over the place and for some reason they are not be
included right and i have to load all of them in the <script> tag in the
html. Don't worry about the late response ,commenting on others work and trying to brainstorm ( is at least
for me) just for fun and  there no rush i not even sure if it helpful. I don't want you to ever feel you
are obligated to respond. I am hard to deal with because I think
differently than most ,all i want to do is be useful.
